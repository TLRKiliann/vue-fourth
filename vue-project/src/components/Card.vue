<script lang="ts">
import { defineComponent } from 'vue';
export default defineComponent({
  // eslint-disable-next-line vue/multi-word-component-names
  name: 'Card',
  data() {
    return {
    }
  },
})
</script>

<template>
    <slot>Default Slot Card</slot>

    <div class="card--header">
        <slot name="header"></slot>
    </div>
    <div class="card--main">
        <slot name="main"></slot>
    </div>
    <div class="card--footer">
        <slot name="footer"></slot>
    </div>
</template>

<style scoped>
.card--header {

  border: 1px solid #fff;
}
.card--main {

border: 1px solid #fff;
}
.card--footer {

border: 1px solid #fff;
}
</style>
