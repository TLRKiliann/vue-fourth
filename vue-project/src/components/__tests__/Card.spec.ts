import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import Card from '../Card.vue'

describe('Card', () => {
  it('snapshot UI testing', () => {
    const wrapper = mount(Card,{});
    expect(wrapper.text()).toMatchSnapshot()
  })
})