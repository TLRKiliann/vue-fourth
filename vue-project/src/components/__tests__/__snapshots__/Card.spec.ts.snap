// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > snapshot UI testing 1`] = `"Default Slot Card"`;
